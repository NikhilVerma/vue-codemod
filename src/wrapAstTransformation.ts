import type { JSCodeshift, Transform, Core, Options } from 'jscodeshift'

export type Context = {
  root: ReturnType<Core>
  j: JSCodeshift
  filename: string
}

export type ASTTransformation = {
  (context: Context, params?: Options): void
}

export default function astTransformationToJSCodeshiftModule(transformAST: ASTTransformation): Transform {
  const transform: Transform = (file, api, options: Options) => {
    const j = api.jscodeshift
    const root = j(file.source)

    transformAST({ root, j, filename: file.path }, options)

    return root.toSource({ lineTerminator: '\n' })
  }

  return transform
}
