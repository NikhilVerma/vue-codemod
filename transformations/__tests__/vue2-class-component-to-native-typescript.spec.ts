import { defineSnapshotTest } from 'jscodeshift/src/testUtils'
import transform from '../vue2-class-component-to-native-typescript'
import fs from 'fs'
import path from 'path'

const testComponent = fs.readFileSync(
  path.resolve(
    __dirname,
    '../__testfixtures__/vue2-class-component-to-native-typescript/TestComponent.input.ts'
  ),
  'utf8'
)

defineSnapshotTest(
  transform,
  { parser: 'tsx' },
  testComponent,
  'removes vue-class-component'
)
