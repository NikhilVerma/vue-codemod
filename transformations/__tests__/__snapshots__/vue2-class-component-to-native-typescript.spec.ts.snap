// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`removes vue-class-component 1`] = `
"import TestComponent1 from './TestComponent1/TestComponent1.vue'
import TestComponent2 from './TestComponent2/TestComponent2.vue'
import { ExampleType } from './ExampleType'

import { defineComponent, PropType } from \\"vue\\";

type ABC = { a: string }
type StrUnion = 'abc' | 'bcd'
type Mode = unknown
type environment = unknown
type project = unknown
type someType = unknown

export default defineComponent({
  name: 'MyComponent',

  components: {
    TestComponent1,
    TestComponent2,
  },

  provide() {
    return {
      foo: this.foo,
      bar: this.baz,
      showModal: this.provideMethod
    };
  },

  inject: {
    foo: \\"foo\\",
    bar: \\"bar\\",
    optional: \\"optional\\",
    baz: \\"baz\\",
    injectMethod: \\"injectMethod\\",
    optional: \\"optional\\"
  },

  props: {
    // @Prop arrTypeSimpleDecorator?: ExampleType[];
    arrTypeSimpleDecorator: {
      type: Array as PropType<ExampleType[]>
    },

    // @Prop({ default: undefined }) undefinedDefault?: project;
    undefinedDefault: {
      type: Object as PropType<project>,
      default: () => undefined
    },

    // @Prop({ default: true }) propWithDefault!: ExampleType[];
    propWithDefault: {
      type: Array as PropType<ExampleType[]>,
      default: () => true
    },

    // @Prop() forcedString!: string;
    forcedString: {
      type: String
    },

    // @Prop({ default: 0 }) numberWithDefault!: number;
    numberWithDefault: {
      type: Number,
      default: () => 0
    },

    // @Prop({ required: true }) unionTypeRequired!: someType | null;
    unionTypeRequired: {
      type: Object as PropType<someType | null>,
      required: true
    },

    // @Prop({ type: Array, required: true }) arrayTypeRequired!: environment[]
    arrayTypeRequired: {
      type: Array as PropType<environment[]>,
      required: true
    },

    // @Prop({ required: true }) stringUnionRequired!: \\"project\\" | \\"environment\\";
    stringUnionRequired: {
      type: String as PropType<'project' | 'environment'>,
      required: true
    },

    // @Prop({ required: true }) complexUnionRequired!: { b: 6 } | ABC;
    complexUnionRequired: {
      type: Object as PropType<{ b: 6 } | ABC>,
      required: true
    },

    // @Prop({ required: false, type: String, default: \\"+ ADD NEW\\" }) optionalStringWithDefault!: string;
    optionalStringWithDefault: {
      type: String,
      default: () => '+ ADD NEW'
    },

    // @Prop({ required: true, type: String }) strUnion!: StrUnion;
    strUnion: {
      type: String as PropType<StrUnion>,
      required: true
    },

    validationProp: {
      type: String as PropType<Mode>,
      default: () => 'aggressive',

      validator(value: string): boolean {
        return ['aggressive', 'passive'].indexOf(value) !== -1
      }
    },

    // @Prop({ required: true, type: Function }) fnProp!: (arg: str) => void;
    fnProp: {
      type: Function as PropType<(arg: string) => void>,
      required: true
    },

    // @Prop({ required: true, type: Function }) fnProp!: (arg: str) => void;
    fnPropDetect: {
      type: Function as PropType<(arg: string) => void>,
      required: true
    },

    width: {
      type: String,
      default: () => 'auto'
    }
  },

  data: () => ({
    data1: true,
    nullable: null as string | null,
    foo: 'foo',
    baz: 'bar'
  }),

  computed: {
    computed1(): boolean {
      return !!this.prop2
    },

    someProp(): string {
      return someStore.someValue
    }
  },

  watch: {
    prop1: {
      deep: true,
      immediate: true,

      handler() {
        console.log('Example watcher')
      }
    }
  },

  created() {
    console.log('created hook')
  },

  methods: {
    provideMethod(target: string, scopeIds?: string[]) {
      const modal = this.$refs.tableModal as AddVariableModal
      modal.openDialog(target, scopeIds ?? [])
    },

    method1(): number {
      console.log('Example method')
    },

    method2(): string {
      console.log('Example method')
    }
  }
});"
`;
